python-package/xgboost/core.py

```diff
@@ -337,6 +338,24 @@ def _maybe_pandas_label(label):
     return label
 
 
+def _maybe_cudf_dataframe(data, feature_names, feature_types):
+    if not (CUDF_INSTALLED and isinstance(data,
+                                          (CUDF_DataFrame, CUDF_Series))):
+        return data, feature_names, feature_types
+    if feature_names is None:
+        if isinstance(data, CUDF_Series):
```

Comment:
We may want to extend this function to support multi-indexes as currently pandas does. (So some porting their pandas code to cudf might face problems . )

We can approach this later but still might be good to throw a error stating that multi-index is not supported.

Reply:
@VibhuJawa Let me get it done in this PR.